Source/SwiftLintCore/Models/Linter.swift

```diff
@@ -62,8 +62,13 @@ private extension Rule {
               configuration: Configuration,
               superfluousDisableCommandRule: SuperfluousDisableCommandRule?,
               compilerArguments: [String]) -> LintResult? {
+        // We shouldn't lint if the current Swift version is not supported by the rule
+        guard SwiftVersion.current >= Self.description.minSwiftVersion else {
+            return nil
+        }
+
         // Empty files shouldn't trigger violations
-        guard !file.isEmpty, SwiftVersion.current >= Self.description.minSwiftVersion else {
+        guard !shouldLintEmptyFiles, !file.isEmpty else {
```

Comment:
This would be easier to grasp with an `if` statement:
```suggestion
        if shouldLintEmptyFiles || file.isEmpty {
```

But then, apparently, nothing is linted when the rule "should lint empty files". That doesn't seem to be correct.